File: abc/087/b/cpp/main.cpp
#include <iostream>

using namespace std;

int main()
{
    int coin_500, coin_100, coin_50, total;
    cin >> coin_500 >> coin_100 >> coin_50 >> total;

    int count = 0;
    for (int i = 0; i <= coin_500; i++)
    {
        for (int j = 0; j <= coin_100; j++)
        {
            for (int k = 0; k <= coin_50; k++)
            {
                if (i * 500 + j * 100 + k * 50 == total)
                {
                    count++;
                }
            }
        }
    }

    cout << count << endl;
}

File: abc/087/b/go/main.go
package main

import (
	"fmt"
)

func main() {
	var coin500, coin100, coin50, total int
	fmt.Scan(&coin500, &coin100, &coin50, &total)

	count := 0
	for i := 0; i <= coin500; i++ {
		for j := 0; j <= coin100; j++ {
			for k := 0; k <= coin50; k++ {
				if i*500+j*100+k*50 == total {
					count++
				}
			}
		}
	}
	fmt.Println(count)
}
